### DISCUSSION TITLE
attempt restart every 15 mins for MacOS

### DISCUSSION THREAD

**Discussion by bug-or-feature** [IbcAlpha/IBC#270] [Q&A] [answered]
The script `gatewaystart.sh` includes these lines

```
if [[ -n $(/usr/bin/pgrep -f "java.*${IBC_INI}") ]]; then
	>&2 echo -e "Error: process is already running"
	>&2 exit 1
fi
```

I'd like similar functionality in MacOS, but the lines are not in the equivalent script. I just did a quick test, adding the lines into my own installed copy (MacOS 13.7 Ventura). When IBC was already running, it carried on running with no issues. When it wasn't, it started up successfully.

Any reason those lines are not present in `gatewaystartmacos.sh` also?

**Comment by rlktradewright** [comment, marked as answer]
Thanks for confirming that this works on macOS. I'll push an update to the macOS script shortly, and include it in the next release.

As to why it wasn't already included, it was probably either an oversight on my part, or perhaps I couldn't ascertain whether the version of Bash in use in macOS at the time actually supported pgrep.

The trouble is I don't have access to a macOS machine, and it certainly used to be the case that the macOS Bash version was well behind the Linux version so I've always been dependent on third parties to test any script updates on macOS.

**Comment by bug-or-feature** [comment]
Great, thank you.

Can I make a suggestion though? Instead of returning 1, return 0. Otherwise `launchd` thinks there has been an error. My script looks like:

```
if [[ -n $(/usr/bin/pgrep -f "java.*${IBC_INI}") ]]; then
	>&2 echo -e "Warning: IBC start attempt aborted, process is already running"
	>&2 exit 0
fi
```

**Comment by rlktradewright** [comment]
Your wish is my cooperation!

**Comment by rlktradewright** [comment]
Update pushed

**Comment by bug-or-feature** [comment]
Awesome! If you like I'll create a new section in the user guide on scheduled running with MacOS, with an example launchd config file

**Comment by rlktradewright** [comment]
Thank you for the offer. That would be very helpful.
